src/vma/dev/ring_tap.cpp

```diff
@@ -371,6 +371,9 @@ int ring_tap::prepare_flow_message(vma_msg_flow& data, msg_flow_t flow_action,
 {
 	int rc = 0;
 
+	if (unlikely(g_p_agent == NULL))
+		return 0;
+
```

Comment:
it is needless and incorrect because conflict with
```
	/* create egress rule (redirect traffic from tap device to physical interface) */
	rc = prepare_flow_message(data, VMA_MSG_FLOW_ADD);
	if (rc != 0) {
		ring_logwarn("Add TC rule failed with error=%d", rc);
	}
```
ring_tap() does not work correctly w/o vma daemon.
So preferable way is to report error once ring_tap() is called. It can be before call or inside constructor.